dumpling/export/ir_impl.go

```diff
@@ -370,3 +371,22 @@ func (td *multiQueriesChunk) Close() error {
 func (*multiQueriesChunk) RawRows() *sql.Rows {
 	return nil
 }
+
+var serverSpecialComments = map[version.ServerType][]string{
```

Comment:
Are these comments for dump files or queries?

Reply:
seems they are written to dump files. These comments take effects when importing not dumping, so we should not using upstream database to decide comments

Reply:
These comments are used for dumped data files. They are used for scenes that we directly restore sqls back through MySQL cmd.
For import tools, they need to re-decide the used comments. These changes are copied from mydumper.

Reply:
need an integration test with lightning later to check we can handle mariadb case😂

Reply:
https://github.com/pingcap/tiflow/blob/099da1e08f1310c2366093f2c314a51e0cde772b/dm/loader/loader.go#L1218

DM loader may not be able to handle `SET FOREIGN_KEY_CHECKS=0;`, not sure what's the behaviour for lightning

Reply:
I think Lightning shouldn't read these comments😂 It should add its own comments according to downstream database.

Reply:
> 

I think DM loader's handle is correct. Loader should execute foreign key handlement itself when it finds its downstream is TiDB. See https://github.com/mydumper/mydumper/blob/e4d05b1a76b7a529d24c43a4d043790f355c5085/src/myloader.c#L388